Cisco/cisco-ftd/ingest/parser.yml

```diff
@@ -0,0 +1,201 @@
+name: cisco-ftd
+pipeline:
+  - name: pre_parsing
+    external:
+      name: grok.match
+      properties:
+        input_field: "original.message"
+        output_field: pre_message
+        pattern: '(<[0-9]+>)?(%{SYSLOGTIMESTAMP:timestamp}|%{TIMESTAMP_ISO8601:timestamp})?(%{IP:log_ip})?(%{SPACE}%{DATA:log_hostname}%{SPACE})?(\:)?\%(FTD|ASA|NGIPS)\-[0-9]{1}\-%{MESSAGE_NUMBER:message_number_grok}\: %{DATA:raw_message}'
```

Comment:
The first part of the grok seems handling syslog header. However, the parser should not face the syslog header (I think you should remove it from the tests).
I suggest:
```suggestion
        pattern: '\%%{(FTD|ASA|NGIPS):product}\-[0-9]{1}\-%{MESSAGE_NUMBER:message_number_grok}\: %{DATA:raw_message}'
```
⚠️ I'm not sure about the `%{(FTD|ASA|NGIPS):product}` syntax